### PR TITLE
Fix some resources that are never closed, close in a finally block

### PR DESCRIPTION
**Pull request by MicrocontrollersDev**
https://github.com/MinecraftForge/MinecraftForge/pull/3637

### CODE REVIEW

File: src/main/java/net/minecraftforge/fml/common/FMLModContainer.java [hunk 1]
```diff
@@ -215,16 +217,24 @@ public Properties searchForVersionProperties() {
                 ZipEntry versionFile = source.getEntry("version.properties");
                 if (versionFile != null) {
                     version = new Properties();
-                    version.load(source.getInputStream(versionFile));
+                    InputStream sourceInputStream = source.getInputStream(versionFile);
+                    try {
+                        version.load(sourceInputStream);
+                    } finally {
+                        IOUtils.closeQuietly(sourceInputStream);
```
Comment (ChachyDev) [inline src/main/java/net/minecraftforge/fml/common/FMLModContainer.java:224]: Maybe you could use a try-catch with resources instead?

File: src/main/java/net/minecraftforge/fml/common/FMLModContainer.java [hunk 2]
```diff
@@ -215,16 +217,24 @@ public Properties searchForVersionProperties() {
...
                 File propsFile = new File(getSource(), "version.properties");
                 if (propsFile.exists() && propsFile.isFile()) {
                     version = new Properties();
                     FileInputStream fis = new FileInputStream(propsFile);
-                    version.load(fis);
-                    fis.close();
+                    try {
+                        version.load(fis);
+                    } finally {
+                        IOUtils.closeQuietly(fis);
```
Comment (ChachyDev) [inline src/main/java/net/minecraftforge/fml/common/FMLModContainer.java:236]: Same applies here

File: src/main/java/net/minecraftforge/fml/common/Loader.java [hunk 3]
```diff
@@ -172,7 +174,12 @@ private Loader() {
         }
 
         minecraft = new MinecraftDummyContainer(MC_VERSION);
-        mcp = new MCPDummyContainer(MetadataCollection.from(getClass().getResourceAsStream("/mcpmod.info"), "MCP").getMetadataForId("mcp", null));
+        InputStream mcpModInputStream = getClass().getResourceAsStream("/mcpmod.info");
+        try {
+            mcp = new MCPDummyContainer(MetadataCollection.from(mcpModInputStream, "MCP").getMetadataForId("mcp", null));
+        } finally {
+            IOUtils.closeQuietly(mcpModInputStream);
```
Comment (ChachyDev) [inline src/main/java/net/minecraftforge/fml/common/Loader.java:181]: Same applies here

File: src/main/java/net/minecraftforge/fml/common/asm/FMLSanityChecker.java [hunk 4]
```diff
@@ -87,7 +88,12 @@ public Void call() throws Exception {
                 mcJarFile = new JarFile(mcPath, true);
                 mcJarFile.getManifest();
                 JarEntry cbrEntry = mcJarFile.getJarEntry("net/minecraft/client/ClientBrandRetriever.class");
-                ByteStreams.toByteArray(mcJarFile.getInputStream(cbrEntry));
+                InputStream mcJarFileInputStream = mcJarFile.getInputStream(cbrEntry);
+                try {
+                    ByteStreams.toByteArray(mcJarFileInputStream);
+                } finally {
+                    IOUtils.closeQuietly(mcJarFileInputStream);
```
Comment (ChachyDev) [inline src/main/java/net/minecraftforge/fml/common/asm/FMLSanityChecker.java:95]: Same applies here

File: src/main/java/net/minecraftforge/fml/common/asm/FMLSanityChecker.java [hunk 5]
```diff
@@ -103,13 +109,7 @@ public Void call() throws Exception {
...
                 FMLRelaunchLog.log(Level.ERROR, e, "A critical error occurred trying to read the minecraft jar file");
             } finally {
-                if (mcJarFile != null) {
-                    try {
-                        mcJarFile.close();
-                    } catch (IOException ioe) {
-                        // Noise
-                    }
-                }
+                IOUtils.closeQuietly(mcJarFile);
```
Comment (ChachyDev) [inline src/main/java/net/minecraftforge/fml/common/asm/FMLSanityChecker.java:112]: Same applies here

**Review by ChachyDev — CHANGES_REQUESTED** [review]

File: src/main/java/net/minecraftforge/fml/common/Loader.java (hunk 3 above)
Comment (MicrocontrollersDev) [reply src/main/java/net/minecraftforge/fml/common/Loader.java:181]: trying to try with resource this caused mcp on line 140 to not be initialized. tbh i didnt even know try with resources was a thing till yesterday so there's a good chance i did this all wrong so

Comment (ChachyDev) [reply src/main/java/net/minecraftforge/fml/common/Loader.java:181]: Hmmm, I would just leave that for now then

**Pull request merged**